src/tt.h

```diff
@@ -21,102 +21,82 @@
 
 #include <cstddef>
 #include <cstdint>
+#include <tuple>
 
 #include "memory.h"
-#include "misc.h"
 #include "types.h"
 
 namespace Stockfish {
 
-// TTEntry struct is the 10 bytes transposition table entry, defined as below:
+class ThreadPool;
+struct TTEntry;
+struct Cluster;
+
+// The Stockfish TranspositionTable is a classic hash table, and some care is taken under the hood to support large hash
+// sizes and efficient storage packing. There is only one global hash table for the engine and all its threads.
+// For chess in particular, we even allow racy updates between threads to and from the TT, as taking the time to
+// synchronize access would cost thinking time and thus elo. As a hash table, collisions are possible and may cause
+// chess playing issues (bizarre blunders, faulty mate reports, etc). However the risk of such decreases with size.
 //
-// key        16 bit
-// depth       8 bit
-// generation  5 bit
-// pv node     1 bit
-// bound type  2 bit
-// move       16 bit
-// value      16 bit
-// eval value 16 bit
+// The public interface of the TT is minimal.
 //
-// These fields are in the same order as accessed by TT::probe(), since memory is fastest sequentially.
-// Equally, the store order in save() matches this order.
-struct TTEntry {
-
-    Move  move() const { return Move(move16); }
-    Value value() const { return Value(value16); }
-    Value eval() const { return Value(eval16); }
-    Depth depth() const { return Depth(depth8 + DEPTH_ENTRY_OFFSET); }
-    bool  is_pv() const { return bool(genBound8 & 0x4); }
-    Bound bound() const { return Bound(genBound8 & 0x3); }
-    void  save(Key k, Value v, bool pv, Bound b, Depth d, Move m, Value ev, uint8_t generation8);
-    // The returned age is a multiple of TranspositionTable::GENERATION_DELTA
-    uint8_t relative_age(const uint8_t generation8) const;
-
+// The maintenance methods are `resize`, `clear` and `hashfull`. The first two do what they say, while `hashfull` reports
```

Comment:
reduce comment to 2-3 lines

Reply:
er, you mean the maintenance method, or you mean all methods combined?!